### ISSUE TITLE
$fireMess undefined in production mode

### ISSUE THREAD

**Issue by sts-ryan-holton** [nuxt-community/firebase-module#172]
### Version
@nuxtjs/firebase: ^5.0.7
firebase: ^7.14.2
nuxt: 2.11.0

### Reproduction Link

https://firebase.nuxtjs.org/guide/usage/#general-usage

### Steps to reproduce

1. Setup Nuxt JS project and install Firebase and the Firebase Module, set to universal mode.
2. Setup config of module with the following services object:

```js
services: {
  messaging: true
}
```

3. The following code is added for token refresh:

```js
var messaging = this.$fireMess;

messaging.onTokenRefresh(() => {
  ...
});
```

### What is Expected?

`this.$fireMess` should be defined in production Nuxt JS.

### What is actually happening?

`this.$fireMess` is **undefined** in production mode of Nuxt JS.

**Comment by lupas** [comment]
hi @sts-ryan-holton 

Firebase Messaging is only loaded on the client, not on the server. Is your code by any change executed e.g. in a plugin or on a hook that is also called on the server?

If so, try wrapping your code in something like this:

```js
if (process.browser) {
   this.$fireMess.onTokenRefresh(() => {}
}
```

**Comment by sts-ryan-holton** [comment]
@lupas My project is running in Universal mode, compiled into HTML using the `build` and `generate` commands.

I'm then doing:

```js
export default {
  mounted() {
    this.initPushNotifications()
  },
  methods: {

    initPushNotifications() {
      var messaging = this.$fireMess;

      messaging.onTokenRefresh(() => {
        messaging.getToken().then((refreshedToken) => {
          ...
        }).catch((err) => {
          ...
        });
      });
    }

  }
}
```

So this works in `npm run dev`, `messaging` returns the object, but not in production mode.

**Comment by lupas** [comment]
@sts-ryan-holton 

So I just did the following:

```js
  mounted() {
    console.log("FIRE MESSAGING", this.$fireMess)
  },
```

* `nuxt build` and then
* `nuxt start`

And it wasn't undefined __in production mode__:
<img width="1440" alt="Screenshot 2020-04-24 at 23 24 40" src="https://user-images.githubusercontent.com/4020037/80258175-dd023000-8682-11ea-9be2-07c3969284d3.png">

So to see if a generated & deployed Nuxt generate behaves the same way, I deployed a static site to Firebase:

```bsh
yarn run generate && firebase deploy --only hosting --project nuxt-fire-demo
```
[https://nuxt-fire-demo.web.app/](https://nuxt-fire-demo.web.app/)
Also works...

So there must be something off with your config, or your environment (e.g. your browser) where you test it. Know that messaging is not supported on all browsers, so it doesn't get initialized if the browser doens't support it.

Though I assume you tested it on the same browser in dev and prod so that's probably not it either...

Btw here is the [branch of the demo](https://github.com/lupas/nuxt-fire-demo/tree/messagingTest) on which I tested it out, maybe the code there helps you to find the issue.

**Comment by sts-ryan-holton** [comment]
`nuxt start` doesn't technically reproduce that of a traditional web server, you're absolutely right, those commands work perfectly fine for me, but try building and then run `npm run generate` or `nuxt generate` afterwards.

I've copied your configuration, and after seeing the generated HTML and checking the console, I get:

`FIRE MESSAGING undefined`

**Comment by lupas** [comment]
> `nuxt start` doesn't technically reproduce that of a traditional web server

You're right, but it runs nuxt in production mode, which is what you named as the issue, so I used it to quickly check if it's just that. 

> but try building and then run npm run generate or nuxt generate afterwards.

If you read my message you see that after that I did __nuxt generate__ and deployed the static html, and did not run into the issue.  So a productive generate deployment works, too.

So that way I could assume it's something related to your setup.

In any case, the information that you run into troubles running it with MAMP Pro is valuable. So I installed MMAP Pro, set up a host, ran it, and: Got same issue as you. Finally =)

Now MAMP Pro doesn't run any SSL just like that, which is [needed for Firebase Messaging to work](https://stackoverflow.com/questions/40298292/does-my-web-application-require-ssl-for-firebase-cloud-messaging-for-web-to-work).

So I did:

1. Host -> SSL
2. Create self-signed certificate..
3. Add something random

Restarted the server and it worked.

Let me know if that does anything for you.
